utoipa-gen/src/path/response.rs

```diff
@@ -64,24 +68,42 @@ impl Parse for ResponseValue<'_> {
 
             match attribute_name {
                 "status" => {
-                    response.status_code =
-                        parse_utils::parse_next(input, || {
-                            let lookahead = input.lookahead1();
-                            if lookahead.peek(LitInt) {
-                                input.parse::<LitInt>()?.base10_parse()
-                            } else if lookahead.peek(LitStr) {
-                                let value = input.parse::<LitStr>()?.value();
-                                if !VALID_STATUS_RANGES.contains(&value.as_str()) {
-                                    return Err(Error::new(
-                                        input.span(),
-                                        format!("{} {}", INVALID_STATUS_RANGE_MESSAGE, VALID_STATUS_RANGES.join(", ")),
-                                    ))
-                                }
-                                Ok(value)
-                            } else {
-                                Err(lookahead.error())
+                    response.status_code = parse_utils::parse_next(input, || {
+                        let lookahead = input.lookahead1();
+                        if lookahead.peek(LitInt) {
+                            input.parse::<LitInt>()?.base10_parse()
+                        } else if lookahead.peek(LitStr) {
+                            let value = input.parse::<LitStr>()?.value();
+                            if !VALID_STATUS_RANGES.contains(&value.as_str()) {
+                                return Err(Error::new(
+                                    input.span(),
+                                    format!(
+                                        "{} {}",
+                                        INVALID_STATUS_RANGE_MESSAGE,
+                                        VALID_STATUS_RANGES.join(", ")
+                                    ),
+                                ));
                             }
-                        })?
+                            Ok(value)
+                        } else if lookahead.peek(syn::Ident) {
+                            input
+                                .parse::<ExprPath>()?
+                                .path
+                                .segments
+                                .last()
+                                .and_then(|segment| {
+                                    STATUS_CODES
+                                        .into_iter()
+                                        .find(|&(_, path)| segment.ident == path)
+                                        .map(|&(code, _)| code.to_owned())
```

Comment:
Here if `status_code` was `Cow<'r, str>` we could just use:
```suggestion
                                        .map(|&(code, _)| Cow::Borrowed(code))
```